### PR TITLE
test: component projectexplorer view

### LINKED ISSUE #537: Warnings in UI tests in SimpeTable.spec.js
Run the tests of `SimpleTable`, see warnings.

### PR DESCRIPTION
**Pull request by SietsmaRJ**
- Added testing for the ProjectExplorer view

Closes #530 due to duplication.
Closes #537

### CODE REVIEW

**Review by marikaris — CHANGES_REQUESTED** [review]
Please fix the warning message (I know there are more warning messages in the rest of the tests, not sure how they got there, but they're not supposed to be there):
```
test("error proceed deleting file", async () => {
        const error = new Error("fail");
        api.deleteObject.mockImplementation(() => {
            return Promise.reject(error.message)
        });
        wrapper.vm.selectedFolder = "doesnt-exist";
        wrapper.vm.selectedFile = "doesnot.exists";
        await wrapper.vm.proceedDelete("doesnt-exist/doesnot.exists");
        await wrapper.vm.$nextTick();
        expect(wrapper.vm.errorMessage).toBe(error.message);
    });
```

**Comment by marikaris** [comment]
[test: component projectexplorer view](https://trello.com/c/1r8qTmy0/1111-test-component-projectexplorer-view)

File: ui/src/views/ProjectsExplorer.vue [hunk 1]
```diff
@@ -431,7 +431,7 @@ export default defineComponent({
           this.successMessage = `Successfully deleted file [${file}] from directory [${folder}] of project: [${this.projectId}]`;
         })
         .catch((error) => {
-          this.errorMessage = error;
+          this.errorMessage = `${error}`;
```
Comment (mswertz) [inline ui/src/views/ProjectsExplorer.vue:434]: unneeded change?

**Review by mswertz — COMMENTED** [review]
I am happy for you to merge this is Mariska also approves

**Comment by marikaris** [comment]
> I am happy for you to merge this is Mariska also approves

Yes, I agree. The error string does indeed seem like an unneeded change, but I'm willing to accept it.

File: ui/tests/unit/components/SimpleTable.spec.ts [hunk 2]
```diff
@@ -72,6 +72,8 @@ describe("SimpleTable", () => {
       props: {
         data: data,
         maxWidth: 600,
+        nCols: 7,
```
Comment (clemens-tolboom) [inline ui/tests/unit/components/SimpleTable.spec.ts:75]: I prefer `nCols: Object.keys(data[0]).length,`

File: ui/src/views/ProjectsExplorer.vue (hunk 1 above)
Comment (marikaris) [reply ui/src/views/ProjectsExplorer.vue:434]: I just tried to "fix" this, but this is in fact correct. There was an issue with typing. `error` is of type `Error`, but this error is of type `string`

Comment (clemens-tolboom) [reply ui/src/views/ProjectsExplorer.vue:434]: Isn't error a reserved word?

**Review by marikaris — APPROVED** [review]

**Pull request merged**